### ISSUE TITLE
undefined method `gsub' for nil:NilClass 

### ISSUE THREAD

**Issue by gauravtiwari** [minimagick/minimagick#328]
Hello @janko-m 

Getting this error:  undefined method `gsub' for nil:NilClass  - mini_magick-4.3.0/lib/mini_magick/shell.rb:32:in `execute'

The error is thrown by the image.resize call (although the same error is thrown by image.valid?, image.blank?, and undoubtedly others), and the line of code in the MiniMagick::Shell class throwing the error is:
``` ruby
send("execute_#{MiniMagick.shell_api.gsub("-", "_")}", *command)
```
Leading me to the conclusion that MiniMagick.shell_api is nil when it shouldn't be.

Thanks, G

**Comment by janko-m** [comment]
That's very strange. I changed in the last version that thread-local variables are used instead of instance variables. I'm not able to reproduce the issue, though, could you help me debug it? After you `require "mini_magick"`, `MiniMagick.shell_api` should be "open3".

**Comment by kapso** [comment]
+1 getting the same error

**Comment by paulanunda** [comment]
+1

**Comment by mgz** [comment]
+1

**Comment by janko-m** [comment]
I yanked version 4.3.0, and released 4.3.1, where I reverted the commit that caused this. It was just to make `MiniMagick.with_cli` thread safe, but it's not worth it. I will maybe bring back part of this commit just for `MiniMagick.cli`, but only when I figure out what the hell happened.
